boards/shared/Inc/App/App_BinaryStatus.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <stdbool.h>
+
+struct BinaryStatus;
+
+/**
+ * Allocate and initialize a binary status indicator
+ * @param is_status_active A function that returns a boolean indicating whether
+ * the binary status is active or inactive
+ * @return The created binary status, whose ownership is given to the caller
+ */
+struct BinaryStatus *App_BinaryStatus_Create(bool (*is_status_active)(void));
```

Comment:
The file name should be `App_SharedBinaryStatus.{c|h}` and the funtion names should be `App_SharedBinaryStatus`.

Reply:
:+1: